src/echo-common/src/cached-task.ts

```diff
@@ -3,84 +3,110 @@ import { throttleTime } from 'rxjs/operators'
 import { filterNullish } from 'ts-ratchet'
 import { EchoDirService } from './echo-dir-service'
 
-export type CachedTaskEvent<T> = {
+export type SmartCacheLoadMode = "Default" | "CacheOnly" | "NoCache"
```

Comment:
Whats important:

- The api should be able to request data without taking the data from the cache, but it should be possible to use cached data if the request is not older than x second. For example a user action (button press), the user makes a request but I want to use cached data, which are not older than 5 seconds, because that way I can save a request.
- Data should be always cached. The option "CacheOnly" or "NoCache" should not exists, because this should be on the subscribtion level. Other apps plugins should still take an advantage if another plugin request new data but don't want to use the cache for their purposes.